edi/views/edi_exchange_record_views.xml

```diff
@@ -23,28 +23,65 @@
             <form string="EDI Exchange Record">
                 <field name="direction" invisible="1" />
                 <field name="retryable" invisible="1" />
-                <header>
-                    <button
-                        name="action_open_related_record"
-                        type="object"
-                        string="Related record"
-                        attrs="{'invisible': [('res_id', '=', False)]}"
-                    />
-                    <!-- FIXME: this `invisible` domain does not work -->
-                    <button
-                        name="action_open_related_exchanges"
-                        type="object"
-                        string="Related exchanges"
-                        attrs="{'invisible': [('related_exchange_ids', '=', False)]}"
-                    />
-                    <button
-                        name="action_retry"
-                        type="object"
-                        string="Retry"
-                        attrs="{'invisible': [('retryable', '=', False)]}"
-                    />
-                </header>
+                <field name="exchange_file_auto_generate" invisible="1" />
+                <header />
                 <sheet>
+                    <div name="button_box" class="oe_button_box">
+<button
+                            name="action_exchange_send"
+                            type="object"
+                            class="oe_stat_button"
+                            icon="fa-send"
+                            attrs="{'invisible': ['|', ('edi_exchange_state', '!=', 'output_pending'), ('exchange_file_auto_generate', '=', True)]}"
+                            string="Send"
+                        />
+                        <button
+                            name="action_exchange_generate"
+                            type="object"
+                            class="oe_stat_button"
+                            icon="fa-cubes"
+                            attrs="{'invisible': ['|', ('edi_exchange_state', '!=', 'new'), ('exchange_file_auto_generate', '=', True)]}"
+                            string="Generate"
+                        />
+                        <button
```

Comment:
same all around. The only btn that should use `exchange_file_auto_generate` is the generate one.
I mean, when something fails you should rely on the automatic machinery to do the job. And for this - apart from the file generation - the "Retry" btn is enough AFAIS.

Reply:
I see, you are right. Fixing it